### PR TITLE
add tracks page to app section with data fetched from the backend

### LINKED ISSUE #461: Move tracks page to app and use material-ui
part of https://github.com/ahlsunnah/learn-islam-ui/issues/441

### PR DESCRIPTION
**Pull request by abumalick**
# بسم الله الرحمن الرحيم

## What's new in this PR?
- chore: update graphqlTypes
- fix: email is of type citext
- feat: add hasura graphql types generation
- fix: avoid malformed  token
- feat: add tracks page in app section with data dynamically fetched from the backend

fixes #461

لله الحمد والمنة

### CODE REVIEW

File: src/components/modules/Tracks/index.tsx [hunk 1]
```diff
@@ -1,18 +1,54 @@
...
+  isLegacy?: boolean
+  path?: string
+}
+const TracksContainer: React.FC<Props> = ({ isLegacy }) => {
+  const { t, i18n } = useTranslation()
+  const language = i18n.language as Locale
+  const { data, loading, error } = useQuery<TracksQuery, TracksQueryVariables>(TRACKS_QUERY, {
+    variables: { locale: language },
+  })
+  console.log(data, loading, error, language)
```
Comment (zerubeus) [inline src/components/modules/Tracks/index.tsx]: console log

File: src/services/apollo.ts [hunk 2]
```diff
@@ -20,11 +20,12 @@ export default function useApolloClientConfig(token: string) {
   const authLink = useMemo(
     () =>
       setContext((_, { headers }) => {
+        const newHeaders = { ...headers }
+        if (token) {
```
Comment (zerubeus) [inline src/services/apollo.ts:24]: 👍 very good

**Review by zerubeus — APPROVED** [review]
LGTM barakallahu fik

**Comment by abumalick** [comment]
@zerubeus feel free to merge if it is ok for you

**Pull request merged**